### DISCUSSION TITLE
Nach Stromausfall wird HMIP RF USB nicht mehr erkannt

### DISCUSSION THREAD

**Discussion by Turkules** [OpenCCU/OpenCCU#2266] [🇩🇪 Fragen&Antworten] [answered]
EDIT:

**Es läuft wieder alles ich habe in der boot config.txt auf 32 bit umgestellt(ist das eine nachhaltige Lösung?), dennoch kann ich mit sudo pivccu-info weiterhin keine Infos abrufen, was komisch für mich erscheint. Würde mich über jede Hilfe freuen! :)**

Nachdem mein Raspberry 4 unerwartet vom Strom getrennt wurde funktionierte plötzlich der Funksensor nicht mehr.
Benutzen tu ich die RaspberryMatic Docker Version.

Leider kann ich den Befehl sudo pivccu-info ebenfalls nicht mehr ausführen und bekomme folgende Fehlermeldung ausgespuckt:

sudo: pivccu-info: command not found

```
root@raspberrypi:~# uname -a
sudo pivccu-info
uname -m
Linux raspberrypi 6.1.19-v8+ #1637 SMP PREEMPT Tue Mar 14 11:11:47 GMT 2023 aarch64 GNU/Linux
sudo: pivccu-info: command not found
aarch64
```

1. Frage: Was könnte das Problem mit Pivccu sein?
2. Frage: Wie kann ich dem Problem mit dem Funksensor auf die Spur gehen? Was ich auf jeden Fall weiß ist, dass die Kernels nicht liefen (EQ3 Char loop not found).
Danke im Voraus! :)

```
root@raspberrypi:~# sudo service pivccu-dkms start
sudo modprobe eq3_char_loop
modprobe: FATAL: Module eq3_char_loop not found in directory /lib/modules/6.1.19-v8+
root@raspberrypi:~# sudo apt update
sudo apt install raspberrypi-kernel raspberrypi-kernel-headers raspberrypi-bootloader
Hit:1 http://raspbian.raspberrypi.org/raspbian bullseye InRelease
Hit:2 http://archive.raspberrypi.org/debian bullseye InRelease
Get:3 https://download.docker.com/linux/raspbian bullseye InRelease [26.7 kB]
Hit:4 https://deb.nodesource.com/node_16.x bullseye InRelease
Hit:5 https://www.pivccu.de/piVCCU stable InRelease
Fetched 26.7 kB in 1s (24.5 kB/s)
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
All packages are up to date.
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
raspberrypi-bootloader is already the newest version (1:1.20230317-1).
raspberrypi-kernel is already the newest version (1:1.20230317-1).
raspberrypi-kernel-headers is already the newest version (1:1.20230317-1).
The following package was automatically installed and is no longer required:
  bc
Use 'sudo apt autoremove' to remove it.
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
root@raspberrypi:~# sudo apt install pivccu-modules-dkms
sudo sh -c 'echo eq3_char_loop >/etc/modules-load.d/eq3_char_loop.conf'
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
pivccu-modules-dkms is already the newest version (1.0.79).
The following package was automatically installed and is no longer required:
  bc
Use 'sudo apt autoremove' to remove it.
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.

The following NEW packages will be installed:
  pivccu-modules-dkms
0 upgraded, 1 newly installed, 0 to remove and 0 not upgraded.
Need to get 42.2 kB of archives.
After this operation, 0 B of additional disk space will be used.
Get:1 https://www.pivccu.de/piVCCU stable/main armhf pivccu-modules-dkms all 1.0.79 [42.2 kB]
Fetched 42.2 kB in 0s (113 kB/s)
Preconfiguring packages ...
Selecting previously unselected package pivccu-modules-dkms.
(Reading database ... 131413 files and directories currently installed.)
Preparing to unpack .../pivccu-modules-dkms_1.0.79_all.deb ...
Unpacking pivccu-modules-dkms (1.0.79) ...
Setting up pivccu-modules-dkms (1.0.79) ...
Error! The module/version combo: pivccu-1.0.79
is not located in the DKMS tree.

Creating symlink /var/lib/dkms/pivccu/1.0.79/source ->
                 /usr/src/pivccu-1.0.79

DKMS: add completed.
Error! echo
Your kernel headers for kernel 6.1.19-v8+ cannot be found at
/lib/modules/6.1.19-v8+/build or /lib/modules/6.1.19-v8+/source.
You can use the --kernelsourcedir option to tell DKMS where it's located.
Created symlink /etc/systemd/system/multi-user.target.wants/pivccu-dkms.service → /lib/systemd/system/pivccu-dkms.service.
Created symlink /etc/systemd/system/debmatic.service.wants/pivccu-dkms.service → /lib/systemd/system/pivccu-dkms.service.
Created symlink /etc/systemd/system/pivccu.service.requires/pivccu-dkms.service → /lib/systemd/system/pivccu-dkms.service.
root@raspberrypi:~# sudo apt-get install raspberrypi-kernel-headers
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
raspberrypi-kernel-headers is already the newest version (1:1.20230317-1).
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
```

**Comment by jens-maus** [comment, marked as answer]
Du bringst hier was Grundsätzliches durcheinander: RaspberryMatic und piVCCU sind zwei paar Schuhe bzw. zwei komplett voneinander unabhängige Lösungen eine CCU alternativ zur Hersteller Software von eQ3 zu betreiben.

Bitte also erst einmal einlesen in die Materie und sich bewusst machen das du hier im RaspberryMatic Projekt Dinge über piVCCU diskutierst die hier aber nicht hingehören sondern eben ins piVCCU Projekt (wenn du piVCCU einsetzt!)

Insofern ist das hier alles leider wohl fehl am Platz!

**Reply by Turkules** [reply]
> Du bringst hier was Grundsätzliches durcheinander: RaspberryMatic und piVCCU sind zwei paar Schuhe bzw. zwei komplett voneinander unabhängige Lösungen eine CCU alternativ zur Hersteller Software von eQ3 zu betreiben.
> 
> Bitte also erst einmal einlesen in die Materie und sich bewusst machen das du hier im RaspberryMatic Projekt Dinge über piVCCU diskutierst die hier aber nicht hingehören sondern eben ins piVCCU Projekt (wenn du piVCCU einsetzt!)
> 
> Insofern ist das hier alles leider wohl fehl am Platz!

Hi Jens, ich bin noch recht neu in dem Thema und bin daher noch etwas Grün hinter den Ohren. Dennoch kurz zurück zu dem eigentlichen Thema. Ich nutze RaspberryMatic. Nach Absturz des Raspis ging wie beschrieben die Erkennung des Funksensors nicht mehr. Als ich wieder erneut alle Packages (Pivcc-dkms etc.) neu versuchte zu installieren hing es bei sudo modprobe eq3_char_loop wie oben beschrieben. Nach hin und her habe ich in der bootconfig das System auf 32 bit umgestellt und seitdem geht es wieder. Kannst du nachvollziehen, weshalb dieser Schritt letztendlich dazu geführt hat das wieder alles läuft? Ich möchte gerne für das nächste mal verstehen können wo das Problem gelegen haben könnte!

Sorry für den etwas inkohärenten Thread.

**Reply by Baxxy13** [reply]
> RaspberryMatic und piVCCU sind zwei paar Schuhe bzw. zwei komplett voneinander unabhängige Lösungen eine CCU alternativ zur Hersteller Software von eQ3 zu betreiben.

Hier wird aber RaspberryMatic als Docker-Container eingesetzt und das setzt nun mal die Installation der "[pivccu-modules-dkms](https://github.com/alexreinert/piVCCU) kernel modules" voraus.

> weshalb dieser Schritt letztendlich dazu geführt hat das wieder alles läuft?

Liest sich wie [das hier](https://homematic-forum.de/forum/viewtopic.php?f=69&t=78380#p761111).
